javascript/javascript3/week3/app.js

```diff
@@ -0,0 +1,126 @@
+import env from "./env.json" assert { type: "json" };
+
+const btnTag = document.getElementById("submitBtn");
+// const saveBtnTag = document.getElementById("saveBtn");
+const urltag = document.getElementById("url");
+const message = document.getElementById("message");
+const screenshotList = document.getElementById("screenshotList");
+const screenshotTag = document.getElementsByClassName("screenshot-wrapper");
+let screenshotUrl;
+
+// function to get screenshot
+const getData = async (urltag) => {
+  const url = `https://website-screenshot6.p.rapidapi.com/screenshot?url=${urltag}&width=1000&height=800&fullscreen=false`;
+  const options = {
+    method: "GET",
+    headers: {
+      "X-RapidAPI-Key": env.apikey,
+      "X-RapidAPI-Host": "website-screenshot6.p.rapidapi.com",
+    },
+  };
+
+  try {
+    const response = await fetch(url, options);
+    const result = await response.text();
+    console.log(result);
+    return result;
+  } catch (error) {
+    console.error(error);
+  }
+};
+// on button click call getData() to get screen shot
+btnTag.addEventListener("click", async (e) => {
+  e.preventDefault();
+  const data = await getData(urltag.value);
+  screenshotUrl = JSON.parse(data).screenshotUrl;
+  displayScreenShot(screenshotUrl);
+  showPubMessage();
+});
+
+// function to display screenshot from api
+function displayScreenShot(screenshotUrl) {
+  console.log(screenshotUrl);
+  const image = document.querySelector(".screenshot");
+  image.src = screenshotUrl;
+  image.alt = "website-screenshot";
+  const screenshotWrapper = document.querySelector(".screenshot-wrapper");
+  const saveButton = document.createElement("button");
+  saveButton.id = "saveBtn";
+  saveButton.innerText = "Save Screenshot";
+  screenshotWrapper.appendChild(saveButton);
+}
+// messages to the user
+function showSavingMessage() {
+  message.innerText = "Your screen shot has saved to server";
+}
+function showPubMessage() {
+  message.innerText = "Please wait... Publishing your screenshot";
+}
+
+// save button click event
+document.addEventListener("click", async (e) => {
+  if (e.target && e.target.id === "saveBtn") {
```

Comment:
Check [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) out. Now you can do this:

```javascript
if (e.target?.id === "saveBtn")
```